### PR TITLE
chore: upgrade to 1.13.0

### PR DESCRIPTION
**Pull request by chungquantin**
Status: ✅ Tested on my local `devnet`, `testnet` => Block production / finalization + submit extrinsic works well. 
- Features: ✅ try-runtime ✅ runtime-benchmarks ✅ metadata-hash
- ISSUE: https://github.com/r0gue-io/pop-node/issues/141
- Release: https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.13.0
- Prerequisite: https://github.com/r0gue-io/pop-node/pull/193

### References
- https://github.com/r0gue-io/base-parachain/pull/38/
- Compare `pallet-contracts`: [`v1.12.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.10.0/substrate/frame/contracts) and [`v1.13.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.11.0/substrate/frame/contracts)
- Compare `pallet-xcm`: [`v1.12.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.10.0/substrate/frame/xcm) and [`v1.13.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.11.0/substrate/frame/xcm)
## Direct changes
- https://github.com/paritytech/polkadot-sdk/pull/4380
- https://github.com/paritytech/polkadot-sdk/pull/3952
- https://github.com/paritytech/polkadot-sdk/pull/4410
- https://github.com/paritytech/polkadot-sdk/pull/4233
- https://github.com/paritytech/polkadot-sdk/pull/4274

## Related changes
- https://github.com/paritytech/polkadot-sdk/pull/4131
- https://github.com/paritytech/polkadot-sdk/pull/4418

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -61,113 +61,112 @@ pop-runtime-common = { path = "runtime/common", default-features = false }
...
+rococo-runtime = { version = "14.0.0", default-features = false }
+xcm = { version = "14.0.0", package = "staging-xcm", default-features = false }
+xcm-builder = { version = "14.0.0", package = "staging-xcm-builder", default-features = false }
+xcm-executor = { version = "14.0.0", package = "staging-xcm-executor", default-features = false }
 
 # Cumulus
-asset-hub-rococo-runtime = { version = "0.17.0", default-features = false }
-asset-test-utils = { version = "13.0.0", default-features = false }
-cumulus-pallet-aura-ext = { version = "0.13.0", default-features = false }
-cumulus-pallet-parachain-system = { version = "0.13.0", default-features = false, features = [
```
Comment (chungquantin) [inline Cargo.toml:148]: https://github.com/paritytech/polkadot-sdk/pull/4380

File: runtime/devnet/src/extensions.rs [hunk 2]
```diff
@@ -124,16 +123,14 @@ where
 	T: pallet_contracts::Config,
 	E: Ext<T = T>,
 {
-	let contract_host_weight = ContractSchedule::<T>::get().host_fn_weights;
-
 	// calculate weight for reading bytes of `len`
-	// reference: https://github.com/paritytech/polkadot-sdk/blob/117a9433dac88d5ac00c058c9b39c511d47749d2/substrate/frame/contracts/src/wasm/runtime.rs#L267
-	let base_weight: Weight = contract_host_weight.return_per_byte.saturating_mul(len.into());
+	// reference: https://github.com/paritytech/polkadot-sdk/pull/4233/files#:~:text=CopyToContract(len)%20%3D%3E%20T%3A%3AWeightInfo%3A%3Aseal_input(len)%2C
+	let base_weight: Weight = T::WeightInfo::seal_return(len);
```
Comment (chungquantin) [inline runtime/devnet/src/extensions.rs:128]: https://github.com/paritytech/polkadot-sdk/pull/4233/files#:~:text=*self%20%7B-,match%20*self%20%7B,CopyFromContract(len)%20%3D%3E,-T%3A%3AWeightInfo

File: runtime/devnet/src/extensions.rs [hunk 3]
```diff
@@ -173,10 +170,7 @@ where
 	let mut env = env.buf_in_buf_out();
 
 	// To be conservative, we charge the weight for reading the input bytes of a fixed-size type.
-	let base_weight: Weight = ContractSchedule::<T>::get()
-		.host_fn_weights
-		.return_per_byte
-		.saturating_mul(env.in_len().into());
+	let base_weight: Weight = T::WeightInfo::seal_return(env.in_len());
```
Comment (chungquantin) [inline runtime/devnet/src/extensions.rs:173]: https://github.com/paritytech/polkadot-sdk/pull/4233/files#:~:text=*self%20%7B-,match%20*self%20%7B,CopyFromContract(len)%20%3D%3E,-T%3A%3AWeightInfo

File: runtime/devnet/src/lib.rs [hunk 4]
```diff
@@ -431,6 +431,10 @@ impl cumulus_pallet_xcmp_queue::Config for Runtime {
 	type ControllerOriginConverter = XcmOriginToTransactDispatchOrigin;
 	type WeightInfo = ();
 	type PriceForSiblingDelivery = NoPriceForMessageDelivery<ParaId>;
+	// Limit the number of messages and signals a HRML channel can have at most
```
Comment (chungquantin) [inline runtime/devnet/src/lib.rs:434]: https://github.com/paritytech/polkadot-sdk/pull/3952

File: runtime/testnet/Cargo.toml [hunk 5]
```diff
@@ -224,3 +225,16 @@ try-runtime = [
 ]
 
 experimental = []
+
```
Comment (chungquantin) [inline runtime/testnet/Cargo.toml:228]: https://github.com/paritytech/polkadot-sdk/pull/4274

**Review by peterwht — APPROVED** [review]
Nicely done

**Comment by peterwht** [comment]
[sc-1154]

**Review by al3mart — CHANGES_REQUESTED** [review]
The only nit would be if we can change the parameters of `enable_metadata_hash` in our `runtime/*/build.rs` to use `PAS` instead of `UNIT`.

**Review by al3mart — APPROVED** [review]

**Pull request merged**